src/plugins/console/public/lib/mappings/mappings.ts

```diff
@@ -0,0 +1,290 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import _ from 'lodash';
+import type { HttpSetup } from '@kbn/core/public';
+import { retrieveSettings, SettingsToRetrieve } from './settings';
+
+interface FieldMapping {
+  enabled?: boolean;
+  path?: string;
+  properties?: Record<string, FieldMapping>;
+  type?: string;
+  index_name?: string;
+  fields?: FieldMapping[];
+}
+
+interface Field {
+  name: string;
+  type: string;
+}
+
+let perIndexTypes: Record<string, object> = {};
+let perAliasIndexes: Record<string, string[]> = {};
```

Comment:
this still feels kind of strange to me, do you think might be better to encapsulate all of these stuff into a class or a constructor function?